src/v2/Utils/Hooks/useDialog.tsx

```diff
@@ -0,0 +1,26 @@
+import React from "react"
+import { useState } from "react"
+
+export interface UseDialogProps {
+  Dialog: React.ComponentType<any>
+  onShow?(): void
+  onHide?(): void
+}
+
+export const useDialog = ({ Dialog, onShow, onHide }: UseDialogProps) => {
```

Comment:
Since we have a few modals and dialogs in Palette, should we move this hook over there?

Reply:
The legacy modal might do this but it's weird and has been really hard to kill. At some point I need to implement a dialog based on `ModalBase` to replace that one in which case this could move over, yeah.